internal/lib/control_config.go

```diff
@@ -125,6 +130,18 @@ func AKOControlConfig() *akoControlConfig {
 	return akoControlConfigInstance
 }
 
+func (c *akoControlConfig) SetIsLeaderFlag(flag bool) {
+	c.isLeaderLock.Lock()
```

Comment:
Yes, locking is required since this variable will be accessed by different go routines such as leader election callbacks, restlayer workers etc within the AKO.